hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDistributionSetManagement.java

```diff
@@ -510,21 +520,23 @@ public DistributionSetMetadata updateDistributionSetMetadata(final DistributionS
         findOne(metadata.getDistributionSet(), metadata.getKey());
         // touch it to update the lock revision because we are modifying the
         // DS indirectly
-        entityManager.merge((JpaDistributionSet) metadata.getDistributionSet()).setLastModifiedAt(0L);
+        final DistributionSet latestDistributionSet = findDistributionSetById(metadata.getDistributionSet().getId());
+        entityManager.merge((JpaDistributionSet) latestDistributionSet).setLastModifiedAt(0L);
         return distributionSetMetadataRepository.save(metadata);
     }
 
     @Override
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
     @Modifying
     public void deleteDistributionSetMetadata(final DistributionSet distributionSet, final String key) {
+        final DistributionSet latestDistributionSet = findDistributionSetById(distributionSet.getId());
```

Comment:
a method for this two of lines would be good, there are multiple used with javadoc.

```
private void touch(final DistributionSet ds)
{
final DistributionSet latestDistributionSet = findDistributionSetById(distributionSet.getId());
entityManager.merge((JpaDistributionSet) latestDistributionSet).setLastModifiedAt(0L);
}